operator/pkg/controllers/etcd/statefulset.go

```diff
@@ -54,6 +54,18 @@ func (c *Controller) reconcileStatefulSet(ctx context.Context, controlPlane *v1a
 				},
 				Spec: etcdSpec,
 			},
+			VolumeClaimTemplates: []v1.PersistentVolumeClaim{{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "etcd-data",
+				},
+				Spec: v1.PersistentVolumeClaimSpec{
```

Comment:
Instead of just exposing StorageClassName and Size, what if we put the entire PersistentVolumeClaimSpec inside the ETCDSpec? Then you could say ETCD.Storage.StorageClassName, ETCD.Storage.Resources or potentially, if inlined, ETCD.StorageClassName and ETCD.Resources (may be a touch confusing with ETCD.Containers[0].Resources)

Reply:
That is exactly what I want to discuss with the team. To me if we are committed to use the AWS-EBS-CSI driver to provision the EBS, then we should definitely expose the entire PersistentVolumeClaimSpec. In addition, we should also dynamically provision storageclass on the fly so that we can specify storagetype and IOPS in the controlplane spec.

On the other hand, I do see that Karpenter can create customized launchtemplate. I need to think a bit more about how to keep Karpenter AWS agnostic.

Reply:
There are many kinds of PVC. It's not tightly coupled to EBS. I expect that KIT will always require some PVC for etc storage (thought it may be RAMFS, EBS, XFS, etc).